src/core/dev/cq_mgr_tx.cpp

```diff
@@ -204,6 +207,27 @@ cq_mgr_tx *cq_mgr_tx::get_cq_mgr_from_cq_event(struct ibv_comp_channel *p_cq_cha
     return p_cq_mgr;
 }
 
+std::string cq_mgr_tx::wqe_to_hexstring(uint16_t index, uint32_t credits) const
+{
+    const auto wqe_begin =
+        static_cast<const uint8_t *>(m_hqtx_ptr->m_mlx5_qp.sq.buf) + WQEBB_SIZE * index;
+
+    std::ostringstream oss;
+    // see `calculate_credits` - credits is give or take the amount of WQEBBs per WQE
+    for (uint32_t wqebb_i = 0; wqebb_i < credits; ++wqebb_i) {
+        const uint32_t wqebb_wrapped_index = wqebb_i & (m_hqtx_ptr->m_tx_num_wr - 1);
```

Comment:
im not sure this will work

Reply:
i think we need to wrap current_wqebb_begin

Reply:
i would say wrap(wqe_begin + wqebb_i * WQEBB_SIZE)

Reply:
This will make the wrapping less readable imho.

The PRM says how to wrap WQEBB index - not actual byte index.

I think we should conform to the PRM reader and not do a complex logic.
Though if you think this doesn't look right, let me know and I'll change.